components/CardStack.jsx

```diff
@@ -0,0 +1,127 @@
+import {
+  motion,
+  useMotionValue,
+  useTransform,
+  useMotionTemplate,
+} from "framer-motion";
+import React, { useState } from "react";
+import Card from "../components/Card";
+import inactiveCards from "../data/inactiveCards";
+
+export const CardStack = ({ vocabularies: vocabulariesprop }) => {
+  const [vocabularies, setVocabularies] = useState(vocabulariesprop);
+  const [inactive, setInactive] = useState(inactiveCards);
+  const [dragStart, setDragStart] = useState({
+    axis: null,
+    animation: { x: 0, y: 0 },
+  });
+  const x = useMotionValue(0);
+  const y = useMotionValue(0);
+  const scale = useTransform(
+    dragStart.axis === "x" ? x : y,
+    [-175, 0, 175],
+    [1, 0.5, 1]
+  );
+  const shadowBlur = useTransform(
+    dragStart.axis === "x" ? x : y,
+    [-175, 0, 175],
+    [0, 25, 0]
+  );
+  const shadowOpacity = useTransform(
+    dragStart.axis === "x" ? x : y,
+    [-175, 0, 175],
+    [0, 0.2, 0]
+  );
+  const boxShadow = useMotionTemplate`0 ${shadowBlur}px 15px -10px rgba(0, 0, 0, ${shadowOpacity})`;
+  const onDirectionLock = (axis) => setDragStart({ ...dragStart, axis: axis });
+  const animateCardSwipe = (animation) => {
+    setDragStart({ ...dragStart, animation });
+
+    setTimeout(() => {
+      setDragStart({ axis: null, animation: { x: 0, y: 0 } });
+      x.set(0);
+      y.set(0);
+      // Removes card if the user swipes left and keeps it if they swipe right //
+      const lastIndex = vocabularies.length - 1;
+      const lastElement = vocabularies[lastIndex];
+      function checkInactive(vocabularies) {
+        return vocabularies.active === false;
+      }
+      function checkActive(vocabularies) {
+        return vocabularies.active === true;
+      }
+      if (animation.x < 0) {
+        lastElement.active = false;
+        const allInactiveCards = vocabularies.filter(checkInactive);
+        const allActiveCards = vocabularies.filter(checkActive);
+        setVocabularies([...allActiveCards]);
+        setInactive([...inactive, ...allInactiveCards]);
+      } else {
+        const newVocabulary = [
+          lastElement,
+          ...vocabularies.slice(0, lastIndex),
+        ];
+        setVocabularies([...newVocabulary]);
+      }
```

Comment:
to me it feels like this would warrant being extracted into separate functions to be easier to read and understand.
Function names are just suggestions and code is not final at all. ;)

```suggestion
      if (animation.x < 0) {
        setCurrentCardInactive()
      } else {
        moveCurrentCardToBottom()
      }
```

Reply:
- [x]  Done!